src/main/java/org/aksw/simba/squirrel/analyzer/impl/RtState.java

```diff
@@ -0,0 +1,48 @@
+package org.aksw.simba.squirrel.analyzer.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * State in a finite state machine.
+ */
+public final class RtState implements State {
+
+    private List<Transition> transitions;
+    private boolean isFinal;
+    private boolean isError;
+
+    public RtState() {
+        this(false, false);
+    }
+
+    public RtState(final boolean isFinal, final boolean isError) {
+        this.transitions = new ArrayList<>();
+        this.isFinal = isFinal;
+        this.isError = isError;
+    }
+
+    public State transit(final String c) {
+        return transitions
+                .stream()
+                .filter(t -> t.isPossible(c))
+                .map(Transition::state)
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("Input not accepted: " + c));
```

Comment:
This is not a good decision. What I see in the code is that you throw an `IllegalArgumentException` but never catch it. In the other classes it looks like you expect that the transition returns an error state if a problem occurs. That would be a better implementation from my point of view.